ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -174,10 +195,13 @@ export function AlertModal({
           <Button
             variant={ButtonVariant.Primary}
             width={BlockSize.Full}
-            onClick={handleButtonClick}
+            onClick={() => {
+              setAlertConfirmed(selectedAlert.key, !isConfirmed);
```

Comment:
My understanding was we want the acknowledgement to be immediate based on the checkbox, and the button is just a means to go to the next one.